src/main/java/rainy/database/Rainy.java

```diff
@@ -0,0 +1,166 @@
+package rainy.database;
+
+import rainy.rainyexceptions.InvalidIndexException;
+import rainy.rainyexceptions.InvalidMarkAndUnmarkException;
+
+import rainy.tasks.TaskTracker;
+
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.IOException;
+
+
+/**
+ * Initializes the Rainy Chatbot.
+ * The Rainy program is a greenfield project building a rudimentary chatbot.
+ *
+ * @author Tan Soo Yap
+ * @since 2024-08-19
+ */
+public class Rainy {
+    /**
+     * Serves as the entry point for the Rainy chatbot.
+     *
+     * @param args  Command-line arguments passed to the program as an array of {@code String} objects.
+     * @throws InvalidIndexException         Thrown by <code>TaskManager</code> object when user provides a non-existent task number.
+     * @throws InvalidMarkAndUnmarkException Thrown by <code>Task</code> object when user wants to mark a marked tasked or unmark an unmarked task.
+     * @throws IOException
+     */
+    public static void main(String[] args) throws InvalidIndexException, InvalidMarkAndUnmarkException, IOException {
+        // Initialize UI, Storage, and TaskTracker, Parser, and File objects
+        UI ui = new UI();
+        Storage storage = new Storage();
+        ui.welcomeMessage();
+        TaskTracker tm = new TaskTracker();
+        Parser ps = new Parser();
+        File newFile = new File("src/main/java/rainy.txt");
+        tm = storage.copyPreviousFiles(newFile);
+        tm.receivedFirstInput();
+
+        // Initialize Scanner object, take in user input
+        Scanner sc = new Scanner(System.in);
+        String messages = sc.nextLine();
+        ps.firstInput(messages);
+        String[] input = ps.getInput();
+        String[] splitByTask = ps.getSplitByTask();
+        String message = ps.getMessage();
+        int count = ps.getCount();
+        Instructions instruction = ps.enumOperator(message);
+
+        // Start processing user inputs
+        while(instruction != Instructions.BYE) {
+            switch (instruction) {
+                case LIST:
+                    System.out.println(tm.getList());
+                    break;
+
+                case MARK:
+                    if (count != -1) {
+                        tm.markDone(count - 1);
+                    } else {
+                        ui.noCategoryDeclared();
+                    }
+                    if (tm.getCounter() > 0) {
+                        File f = new File("src/main/java/rainy.txt");
```

Comment:
Note there is no indentation for case clauses.

Reply:
I've changed it, thank you!!